src/runner/main.cpp

```diff
@@ -375,6 +375,7 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine
         modules();
 
         auto general_settings = load_general_settings();
+        apply_general_settings(general_settings);

```

Comment:
Sorry, I had already started working on the issue, I wasn't aware that you took it on.  The setting is applied but un-done when settings V2 is launched. The reason behind this is that the runner does not save the settings to the JSON config file after applying the "run on startup" configuration. Settings V1 is able to catch this since it asks runner upon launch. To fix this, we need to tell the runner to save updated configurations to the JSON file upon launch since settings V2 checks if the file exists before it creates a new one with default configurations. IMHO, I don't think this would solve the issue #3875. But its close 😀.